### PR TITLE
Make imgui font glyph initializers more compiler and binary size friendly

### PR DESCRIPTION
**Pull request by ehughsbaird**
#### Summary
None

#### Purpose of change
See title.

#### Describe the solution
List the glyphs in an array, then add the glyphs in a for loop.

#### Describe alternatives you've considered
See the first commit.

#### Testing
I don't have a tiles build handy to test.

### CODE REVIEW

**Comment by akrieger** [comment]
I guess it's... not more broken than before?

**Comment by akrieger** [comment]
Ok technically clang-tidy needs another suppression.

**Comment by CLIDragon** [comment]
Lines 13 - 39 could be made into one multiline string which would be much more readable.

File: build-scripts/get_translation_characters.py [hunk 1]
```diff
@@ -14,6 +15,28 @@ def main():
           "el es fr hu id is it ja ko nb nl pl pt ru sr tr uk_UA zh_Hans "
           "zh_Hant > src/cldr/imgui-glyph-ranges.cpp\n")
     print("// NOLINTBEGIN(cata-static-declarations,readability-function-size)")
+    print("")
+    print("#if defined(__GNUC__) or defined(__clang__)")
```
Comment (db48x) [inline build-scripts/get_translation_characters.py:19]: Can you tidy this up by moving the preprocessor nonsense into a separate function?

**Review by db48x — CHANGES_REQUESTED** [review]

**Comment by ehughsbaird** [comment]
It seems good to me.
image showing language selection screen with properly rendered names of languages
image showing the item info page for a fire hydrant in Japanese
image showing the item info page for the same fire hydrant in Russian

**Review by db48x — APPROVED** [review]

**Pull request merged**